### DISCUSSION TITLE
[object object] on every refresh of the page

### DISCUSSION THREAD

**Discussion by jjasghar** [open-webui/open-webui#4837] [General]
# Bug Report

## Installation Method

It seems whenever i open up a previous chat, i see this red box at the bottom:
<img width="1193" alt="image" src="https://github.com/user-attachments/assets/562d4ba5-95cf-4afe-8492-59dd274b5ca8">

## Environment

- **Open WebUI Version:** 0.3.6

- **Operating System:** CentOS 9 stream
- **Browser (if applicable):** Chrome, Safari

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

No box, and if there is an error there something for me to act on.

**Comment by sukirtharajan18** [comment]
is there a fix for this?

**Comment by semidark** [comment]
I have the same problem but with a slightly different setup. I am running open WebUI as Docker and behind an Apache that is doing the SSL Termination. Everything is running on an Ubuntu 20.04 Server. I currently manly use the OpenAI Models directly over the API. 

It is a really annoying Issue, since the chat basically ends when this happens. I haven't found any other workaround but resending the prompt, stopping the response ASAP and then copy and paste the original response so I can work with it again. IMHO it is really messy workaround and I would like to understand the issue.

**Comment by semidark** [comment]
I did the following without any improvement on the issue:

- Cloning the Broken Chat
- Changed the browser from Chromium to Firefox 
- Updated the Open WebUI Docker Container to the latest release v0.3.16
- Create a new User without any History nor Personalisation
  This at least let me work again with the System for 2 Hours. After that the error hit again

Any Ideas what to try next?

**Comment by Shayano** [comment]
I had the same problem but with nginx
You need to change the headers for the websocket on apache.  
I don't know the procedure for apache, but below is an example of a configuration that works for another user.

In my opinion, this should be indicated in the documentation, as it worked in previous versions without this modification.

**Nginx example** (important part is the WebSocket support)
```
location / {
   proxy_pass http://127.0.0.1:8080;
   proxy_set_header Host $host;
   proxy_set_header X-Real-IP $remote_addr;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header X-Forwarded-Proto $scheme;

   # WebSocket support
   **proxy_http_version 1.1;
   proxy_set_header Upgrade $http_upgrade;
   proxy_set_header Connection "upgrade";**
}

```


**Apache example**
```
<VirtualHost *:80>
    ServerName CHANGEME.domain.com

    ServerAdmin webmaster@localhost
    DocumentRoot /var/www/html

    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined

    RewriteEngine on
    RewriteCond %{SERVER_NAME} =CHANGEME.domain.com
    RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>

########################################################################

<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerName CHANGEME.domain.com

    ServerAdmin webmaster@localhost

    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined

    ProxyRequests Off
    ProxyPreserveHost On
    ProxyAddHeaders On
    SSLProxyEngine On

    # ProxyPass et ProxyPassReverse
    ProxyPass / http://CHANGEME.internal:8080/ nocanon
    ProxyPassReverse / http://CHANGEME.internal:8080/

    # Support WebSocket
    RewriteEngine On
    RewriteCond %{HTTP:Upgrade} =websocket [NC]
    RewriteRule /(.*) ws://CHANGEME.internal:8080/$1 [P,L]
    RewriteCond %{HTTP:Upgrade} !=websocket [NC]
    RewriteRule /(.*) http://CHANGEME.internal:8080/$1 [P,L]

    # Headers configuration using SetEnvIf
    SetEnvIf Remote_Addr "(.*)" X_REAL_IP=$1
    RequestHeader set X-Real-IP "%{X_REAL_IP}e"

    RequestHeader append X-Forwarded-For "%{X_REAL_IP}e"

    RequestHeader set X-Forwarded-Proto "https"

    # Specific location configurations
    <Location "/CHANGEME/api">
        ProxyPass http://CHANGEME.internal2:12345/api nocanon
        ProxyPassReverse http://CHANGEME.internal2:12345/api/
    </Location>
    
    <Location "/api/v1">
        ProxyPass http://CHANGEME.internal:8080/api/v1 nocanon
        ProxyPassReverse http://CHANGEME.internal:8080/api/v1
    </Location>

    SSLCertificateFile /etc/ssl/CHANGEME/CHANGEME.crt
    SSLCertificateKeyFile /etc/ssl/CHANGEME/CHANGEME.key
    Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```



Source:
https://github.com/open-webui/open-webui/issues/4158#issuecomment-2262668457

**Reply by semidark** [reply]
Yes, I am using Apache as a reverse proxy to handle TLS traffic. As mentioned in the source you provided, I also noticed that the JavaScript console showed an error about a WebSocket connection. However, I haven't had the time to investigate this further yet. I will apply your valuable input right away and report back with the results. Thanks!

**Comment by semidark** [comment]
Alright, I tried implementing the suggested solution for the Apache server, but I could not get it working. So, I read up on how to handle WebSocket in the [Apache documentation](https://httpd.apache.org/docs/2.4/mod/mod_proxy_wstunnel.html). Here is the configuration that is working for me:

```
<IfModule mod_ssl.c>
    <VirtualHost localhost:443>
        ServerName domain.example.com
	SecRuleEngine Off

        ProxyPreserveHost On
        ProxyPass / http://127.0.0.1:8080/ nocanon
        ProxyPassReverse / http://127.0.0.1:8080/

        # Add WebsSocket Support --->
        RewriteEngine on
        RewriteCond %{HTTP:Upgrade} websocket [NC]
        RewriteCond %{HTTP:Connection} upgrade [NC]
        RewriteRule ^/?(.*) "ws://127.0.0.1:8080/$1" [P,L]
        # <---

        RequestHeader set X-Forwarded-Proto "https"
        RequestHeader set X-Forwarded-Ssl on

        ErrorLog ${APACHE_LOG_DIR}/open-webui-site_error.log
        CustomLog ${APACHE_LOG_DIR}/open-webui_access.log combined

        SSLCertificateFile /etc/letsencrypt/live/domain.example.com/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/domain.example.com/privkey.pem
        Include /etc/letsencrypt/options-ssl-apache.conf
    </VirtualHost>
</IfModule>
```

**Reply by semidark** [reply]
I created a Pull Request for an update to the Documentation: https://github.com/open-webui/docs/pull/199
